repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,284 @@
+        all_repairs = (self | group_repairs).with_context(skip_group_cancel=True)
+
+        # Prevent calling the original method on an empty recordset
+        if not all_repairs:
+            return True
+
+        # Single call to original method to preserve full Odoo workflow
+        return super(RepairOrder, all_repairs).action_repair_cancel()
+
+    def action_create_sale_order(self):
+        """Create Sale Order - single one for group or standard behavior if no group.
+
+        For repair groups: creates one sale order containing all valid repairs
+        from the same group. Maintains core Odoo validations while extending
+        functionality for grouped repairs.
+
+        Features:
+        • Consolidated SO creation for repair groups
+        • Standard individual SO creation for non-grouped repairs
+        • Mixed selection handling with focused UX
+        • Detailed validation messages with repair order references
+
+        Returns:
+            dict: Action to view the created sale order(s)
+        """
+
+        # Split selected repairs into grouped and ungrouped
+        grouped = self.filtered(lambda r: r.group_id)
+        ungrouped = self - grouped
+
+        # No groups at all → fallback to core logic
+        if not grouped:
+            return super().action_create_sale_order()
+
+        # Core validations (same as original Odoo, but applied once)
+        concerned = self.filtered("sale_order_id")
+        if concerned:
+            ref_str = "\n".join(concerned.mapped("name"))
+            raise UserError(
+                _(
+                    "You cannot create a quotation for a repair order that is "
+                    "already linked to an existing sale order.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        no_partner = self.filtered(lambda ro: not ro.partner_id)
+        if no_partner:
+            ref_str = "\n".join(no_partner.mapped("name"))
+            raise UserError(
+                _(
+                    "You need to define a customer for a repair order in order to "
+                    "create an associated quotation.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        created_so_ids = []
+
+        # One Sale Order per unique repair group
+        for group in grouped.mapped("group_id"):
+            existing_so = group.repair_ids.filtered("sale_order_id")
+            if existing_so:
+                ref_str = "\n".join(existing_so.mapped("name"))
+                raise UserError(
+                    _(
+                        "A sale order already exists for this repair group.\n"
+                        "Concerned repair order(s):\n%(ref_str)s",
+                        ref_str=ref_str,
+                    )
+                )
+
+            repairs = group.repair_ids.filtered(
```

Comment:
Line 221 iterates the same recordset again. You can check the order state there and then use the `line_commands` value.

```python
for repair in repairs:
    if repair.state == "cancel":
        continue
....
if not line_commands:
...
```